packages/the-org/src/loadTest/index.ts

```diff
@@ -0,0 +1,552 @@
+        this.scenarioService = runtime.getService('scenario');
+        if (!this.scenarioService) {
+          throw new Error('Scenario service not found');
+        }
+
+        // Ensure logs directory exists
+        const __filename = fileURLToPath(import.meta.url);
+        const __dirname = path.dirname(__filename);
+        const logsDir = path.join(__dirname, 'logs');
+        if (!fs.existsSync(logsDir)) {
+          fs.mkdirSync(logsDir, { recursive: true });
+        }
+
+        const timestamp = new Date().toISOString().replace(/:/g, '-');
+        const summaryLogFile = path.join(logsDir, `load-test-summary-${timestamp}.log`);
+        const systemMetricsFile = path.join(logsDir, `system-metrics-${timestamp}.log`);
+        
+        // Start system metrics monitoring
+        this.startSystemMetricsMonitoring();
+        
+        // Write header to summary file
+        fs.writeFileSync(summaryLogFile, 
+          'Scale Testing Summary - Finding Breaking Points\n' +
+          '===========================================\n' +
+          `Test Started: ${new Date().toISOString()}\n\n` +
+          'Configuration | Agents | Messages | Success Rate | Avg Response | Memory | Throughput | Errors\n' +
+          '-------------|--------|----------|--------------|--------------|--------|------------|-------\n'
+        );
+
+        logger.info('Starting sequential load tests to find breaking points');
+        
+        // Main testing loop - continue until breaking point or all tests complete
+        let breakingPointIdentified = false;
+        
+        for (const config of this.SCALE_CONFIGS) {
+          // Skip if we've already found a breaking point
+          if (breakingPointIdentified) {
+            fs.appendFileSync(summaryLogFile, 
+              `${config.description.padEnd(13)} | ${String(config.agents).padEnd(6)} | ` +
+              `${String(config.messageCount).padEnd(8)} | SKIPPED | - | - | - | -\n`
+            );
+            continue;
+          }
+          
+          const detailedLogFile = path.join(logsDir, `load-test-${config.agents}-agents-${timestamp}.log`);
+          
+          try {
+            // Write test header to detailed log
+            fs.writeFileSync(detailedLogFile, 
+              `Load Test: ${config.description}\n` +
+              `================${'='.repeat(config.description.length)}\n` +
+              `Agents: ${config.agents}\n` +
+              `Messages: ${config.messageCount}\n` +
+              `Started: ${new Date().toISOString()}\n\n` +
+              'DETAILED TEST LOG:\n' +
+              '=================\n\n'
+            );
+            
+            logger.info(`\nStarting load test for ${config.description}`);
+            const metrics = await this.runSequentialLoadTest(runtime, config);
+            
+            // Check if this is a breaking point (high error rate or very slow responses)
+            if (metrics.errorCount > 0.2 * metrics.messagesSent || 
```

Comment:
This is a great addition - let's make these thresholds configurable / easier to discern by making this a constant and also consider running the load tests 3 with 3 different configurations of it i.e. with 3 thresholds like HIGH, MED, LOW and thresholds configured accordingly. That way in a single run we can see BREAKING POINT IDENTIFIED WITH HIGH THRESHOLD(details of thresholds), BREAKING POINT IDENTIFIED WITH MID THRESHOLD (details) , BREAKING POINT IDENTIFIED WITH LOW THRESHOLD (thresholds) lmk if you have any questions